test/karma-conf.js

```diff
@@ -740,3 +680,74 @@ module.exports = function (config) {
 
     config.set(options);
 };
+
+function createVisualTestTemplate(argv, path, js, assertion) {
+    let scriptBody = resolveJSON(js);
+
+    // Don't do intervals (typically for gauge samples, add point etc)
+    scriptBody = scriptBody.replace('setInterval', 'Highcharts.noop');
+
+    // Force enableSimulation: false
+    scriptBody = scriptBody.replace('enableSimulation: true','enableSimulation: false');
+
+    let html = getHTML(path);
+
+    // Reset global options
+    let reset = (
+        scriptBody.indexOf('Highcharts.setOptions') === -1 &&
+        scriptBody.indexOf('Highcharts.getOptions') === -1
+    ) ?
+        '' :
+        `
+        Highcharts.setOptions(
+            JSON.parse(Highcharts.defaultOptionsRaw)
+        );
+        `;
+
+    // Reset modified callbacks
+    if (scriptBody.indexOf('Chart.prototype.callbacks') !== -1) {
+        reset += `
+            Highcharts.Chart.prototype.callbacks =
+                Highcharts.callbacksRaw.slice(0);
+        `;
+    }
+
+    return `
+        QUnit.test('${path}', function (assert) {
+            // Apply demo.html
+            document.getElementById('demo-html').innerHTML = \`${html}\`;
+            
+            /*
+             * we expect 2 callbacks if --visualcompare argument is supplied,
+             * one for the test comparison and one for checking if chart exists.
+             */ 
+            var done = assert.async(${argv.visualcompare || argv.reference ? 2 : 1 });
+            
+            ${scriptBody}
+
+            let attempts = 0;
+            function assertIfChartExists() {
+                var chart = Highcharts.charts[
+                    Highcharts.charts.length - 1
+                ];
+
+                if (chart && document.getElementsByTagName('svg').length) {
```

Comment:
Should be `chart || document.getElementsByTagName('svg').length` in order to allow renderer-only samples.

Reply:
Fixed